### PR TITLE
IoUring: Refactor IoUringIoOps to better match the underlying io_urin…

### PR DESCRIPTION
**Pull request by normanmaurer**
…g_sqe

Modification:

We should be able to support any io_uring operation in the future for this we should allow to setup the right struct.

Modifications:

- Change constructor to match underlying struct
- Remove public getters as these are only used internally and might change in the future
- Rename static fields

Result:

More flexible implementation for io_uring

### CODE REVIEW

File: transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoOps.java [hunk 1]
```diff
@@ -19,137 +19,185 @@
...
+     *          __s32 splice_fd_in;
+     *          __u32 file_index;
+     *          __u32 optlen;
+     *          struct {
+     *              __u16 addr_len;
+     *              __u16 __pad3[ 1];
+     *          };
+     *      };
+     *
+     *      union {           // union6
```
Comment (normanmaurer) [inline transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoOps.java:118]: I used a long for this to keep it simple. If we need something more advanced later we can add another constructor that takes a byte[].

File: transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoOps.java [hunk 2]
```diff
@@ -19,137 +19,185 @@
...
-    private final int spliceFdIn;
+    private final long union1;
+    private final long union2;
+    private final int len;
+    private final int union3;
     private final short data;
+    private final short personality;
+    private final short union4;
+    private final int union5;
+    private final long union6;
```
[the remark this answers is not in the document]
Comment (franz1981) [reply transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoOps.java:38]: memcpy (using Unsafe, I suppose?) are not super well implemented for small amount of bytes unless we're talking about something big, ATM - on JDK at least

[the remark this answers is not in the document]
Comment (normanmaurer) [reply transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoOps.java:38]: yes...

File: transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoOps.java [hunk 3]
```diff
@@ -19,137 +19,185 @@
...
+     *          // this field is used for 80 bytes of arbitrary command data
+     *          __u8 cmd[ 0];
+     *      };
+     *  };
+     * }
+     * </pre>
      */
-    public IoUringIoOps(byte opcode, int flags, short ioPrio, int fd, int rwFlags, long bufferAddress,
-                        int length, long offset, int spliceFdIn, short data) {
+    public IoUringIoOps(byte opcode, byte flags, short ioPrio, int fd, long union1, long union2, int len, int union3,
```
Comment (normanmaurer) [inline transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoOps.java:134]: We might want to consider adding a builder later to make things easier.

Comment (chrisvest) [reply transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoOps.java:134]: Yeah, a builder could have overloads for the unions, so you can set the field with the right name for the context.

Comment (normanmaurer) [reply transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoOps.java:134]: lets do this as a followup tho

**Review by chrisvest — APPROVED** [review]
One build failed with a leak, though.

**Comment by normanmaurer** [comment]
@chrisvest yeah but the leak even happens without this PR for some reason. I am still not sure why

**Pull request merged**